test/junit/scala/collection/mutable/ArrayBufferTest.scala

```diff
@@ -508,4 +508,15 @@ class ArrayBufferTest {
     check(_ ++= (1 to 100))
     check(_.insertAll(1, 1 to 100))
   }
+
+  @Test def `drop right when empty is empty`: Unit = {
+    assertTrue(ArrayBuffer().dropRight(1).isEmpty)
+  }
+
+  @Test def `refuse to build from an overly huge thing`: Unit = {
+    val bld = ArrayBuffer.newBuilder[String]
+    assertThrows[Exception](bld.sizeHint(Int.MaxValue), _.contains("exceeds VM length limit"))
+    bld.addOne("hello, world")
+    assertTrue(bld.result().contains("hello, world"))
+  }
```

Comment:
Missed opportunity to test -1, -100. The other test is for the resize, this is for the builder which delegates to ArrayBuffer. By definition the behavior is undefined. It would also be nice to have tests that probe that size hints are respected, although, also by definition, the user of the API can't easily tell the difference.